cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AdapterHttpHandler.java

```diff
@@ -276,7 +276,7 @@ public void deleteAdapter(HttpRequest request, HttpResponder responder,
       adapterService.removeAdapter(Id.Namespace.from(namespaceId), adapterName);
       responder.sendStatus(HttpResponseStatus.OK);
     } catch (CannotBeDeletedException e) {
-      responder.sendString(HttpResponseStatus.FORBIDDEN, e.getMessage());
+      responder.sendString(HttpResponseStatus.CONFLICT, e.getMessage());
```

Comment:
Are most `CannotBeDeletedException` supposed to result in CONFLICT? If so, you can remove this catch-block and simply catch it in https://github.com/caskdata/cdap/blob/develop/cdap-common/src/main/java/co/cask/cdap/common/exception/HttpExceptionHandler.java#L35